src/test/scala/beam/router/PhyssymXmlToOsmConverter.scala

```diff
@@ -61,48 +64,47 @@ object PhyssymXmlToOsmConverter extends StrictLogging {
               startElement.getAttributeByName(new QName("from")).getValue,
               startElement.getAttributeByName(new QName("to")).getValue
             )
+            wayId = startElement.getAttributeByName(new QName("id")).getValue
             wayInnerTags = buildLinkProps(startElement)
           case "attributes" if canProcessLink =>
             canProcessLinkAttributes = true
           case "attribute" if canProcessLinkAttributes && tagNameHasKind(startElement, "origid") =>
-            canProcessAttribute = true
-            isWayId = true
             isHighway = false
           case "attribute" if canProcessLinkAttributes && tagNameHasKind(startElement, "type") =>
-            isWayId = false
             isHighway = true
           case somethingElse =>
             logger.warn(s"Unidentified tag: [$somethingElse]")
         }
       } else if (nextEvent.isEndElement) {
         val endElement = nextEvent.asEndElement()
         endElement.getName.getLocalPart match {
-          case "nodes" => canProcessNode = false
-          case "link" if canProcessLink =>
-            links += Way(wayId, source_destination._1, source_destination._2, wayInnerTags)
+          case "network"                =>
+          case "nodes"                  => canProcessNode = false
+          case "node" if canProcessNode =>
           case "links" if canProcessLink =>
             canProcessLink = false
-          case "attributes" if canProcessLinkAttributes => canProcessLinkAttributes = false
-          case "attribute" if canProcessLinkAttributes && isWayId =>
-            isWayId = false
-          case "attribute" if canProcessLinkAttributes && isHighway =>
-            isHighway = false
+          case "link" if canProcessLink =>
+            links += Way(wayId, source_destination._1, source_destination._2, wayInnerTags)
+          case "attributes" if canProcessLinkAttributes =>
+            canProcessLinkAttributes = false
+          case "attribute" if canProcessLinkAttributes =>
+            if (isHighway) {
+              isHighway = false
+            }
           case somethingElse =>
-            logger.warn(s"Something not predicted $somethingElse")
+            logger.warn(
+              s"Something not predicted. canProcessLinks: [$canProcessLink]; [canProcessLinkAttributes: [$canProcessLinkAttributes]; canProcessNode: [$canProcessNode]; somethingElse: [$somethingElse]"
+            )
         }
       } else if (nextEvent.isCharacters) {
         val characters: Characters = nextEvent.asCharacters()
         val charactersValue = characters.getData
-        if (!isBlank(charactersValue)) {
-          if (isWayId) {
-            wayId = charactersValue
```

Comment:
This was removed as part of the refactoring you did - was that on purpose? Or is that the bugfix you mention?

Reply:
is the bugfix